lib/containers/docker.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Basic functions for testing docker
+# Maintainer: Anna Minou <anna.minou@suse.de>
+
+package containers::docker;
+use base "consoletest";
+use testapi;
+use utils;
+use strict;
+use warnings;
+use version_utils;
+
+my $my_app = << 'EOF';
+from flask import Flask, render_template
+import os, sys, json, urllib.request, ssl
+
+app = Flask(__name__)
+version = ""
+
+@app.route("/")
+def index():
+    return render_template("index.html", version = version)
+
+if __name__ == "__main__":
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    
+    if len(sys.argv) > 1:
+        url = sys.argv[1]
+    else:
+        print("Program requires at least one argument!")
+        sys.exit()
+
+    request = urllib.request.urlopen(url, context=ctx)
+    data = json.loads(request.read())
+    version = (data["job"]["settings"]["VERSION"])
+    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
+
+EOF
+
+my $dockerfile = << 'EOF';
+FROM python:3
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY . .
+
+# install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# tell the port number the container should expose
+EXPOSE 5000
+
+# run the command
+ENTRYPOINT ["python", "./app.py"]
+EOF
+
+my $requirements = << 'EOF';
+Flask==1.0
+EOF
+
+my $index_html = << 'EOF';
+<html>
+	You shall not pass in version: {{ version }}!
+</html>
+EOF
+
+my $dir = '';
+my $id  = '';
+
+# Setup environment
+sub set_up() {
+    $dir = "/root/DockerTest";
+    assert_script_run("mkdir $dir");
+    assert_script_run("cd $dir");
+    assert_script_run("mkdir BuildTest");
+    assert_script_run("echo '$my_app' >> BuildTest/app.py");
+    assert_script_run("echo '$dockerfile' >> BuildTest/Dockerfile");
+    assert_script_run("echo '$requirements' >> BuildTest/requirements.txt");
+}
+
+# Identify the job id
+
+sub job_id() {
+    my $name = get_var('NAME');
+    $id = (split('-', $name))[0];
+}
```

Comment:
I think you can move this into `test_built_img()` because you use the `$id` only there.

Reply:
I expanded a bit the function, so I let it in the code as : get_vars()